### PR TITLE
naive handle of master

### PR DESCRIPTION
**Pull request by jespada**
This is a naive way of dealing with `riemann-mysql` running on master.
The idea is that if we swap master and a master node becomes slave we automatically start tracking the replication lag.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -208,6 +208,23 @@ func main() {
 					goto send
 				}
 
+				// If
+				// MariaDB [(none)]> show all slaves status;
+				// Empty set (0.000 sec)
+				// we assume is a master
+				if r.Resultset.RowNumber() == 0 {
+					log.Info("There is no replication status, looks like master")
```
Comment (falzm) [inline main.go]: I'm not sure this message requires the `info` level:

```suggestion
					log.Debug("no replication status, looks like master")
```

File: main.go [hunk 2]
```diff
@@ -208,6 +208,23 @@ func main() {
...
+					events = append(events, &raidman.Event{
+						Time:        t.Unix(),
+						Service:     "mysql/replication/master",
+						State:       "ok",
+						Description: "Looks like this is the master",
```
Comment (falzm) [inline main.go]:
```suggestion
						Description: "master OK",
```

😏

**Review by falzm — CHANGES_REQUESTED** [review]
Hey @jespada, thank you for your contribution. Unless I'm mistaken your change doesn't need the `go.{mod,sum}` files to be updated, could you please revert those changes in order to focus your PR on solely the code logic modification and avoid potential dependencies update side-effects?

**Review by falzm — APPROVED** [review]
LGTM 👍

**Review by pyr — COMMENTED** [review]
Wouldn't this also be the behavior if replication stopped?

**Comment by LaCap** [comment]
@pyr yes could be an issue if there is an old configuration or if the replication is only stopped.
This could happen on our instances.

**Comment by jespada** [comment]
> Wouldn't this also be the behavior if replication stopped?

@pyr  @LaCap  hence the naive in the PR title, if thats the case it will be catch by https://github.com/exoscale/riemann-mysql/blob/check_master/main.go#L251
and we'll get a warning, which I think is fine, we go and clean it up if its an old config or we fix the issue.

**Comment by LaCap** [comment]
@jespada even if you clean the configuration the MySQL engine keep some traces...
So the query you perform will still send some rows...
```
MariaDB [(none)]> show slave status\G
*************************** 1. row ***************************
               Slave_IO_State: 
                  Master_Host: HOST
                  Master_User: USER
                  Master_Port: 3306
                Connect_Retry: 60
              Master_Log_File: 
          Read_Master_Log_Pos: 4
               Relay_Log_File: 
                Relay_Log_Pos: 691
        Relay_Master_Log_File: 
             Slave_IO_Running: No
            Slave_SQL_Running: No
              Replicate_Do_DB: 
          Replicate_Ignore_DB: 
           Replicate_Do_Table: 
       Replicate_Ignore_Table: 
      Replicate_Wild_Do_Table: 
  Replicate_Wild_Ignore_Table: 
                   Last_Errno: 0
                   Last_Error: 
                 Skip_Counter: 0
          Exec_Master_Log_Pos: 0
              Relay_Log_Space: 993
              Until_Condition: None
               Until_Log_File: 
                Until_Log_Pos: 0
           Master_SSL_Allowed: No
           Master_SSL_CA_File: 
           Master_SSL_CA_Path: 
              Master_SSL_Cert: 
            Master_SSL_Cipher: 
               Master_SSL_Key: 
        Seconds_Behind_Master: NULL
Master_SSL_Verify_Server_Cert: No
                Last_IO_Errno: 0
                Last_IO_Error: 
               Last_SQL_Errno: 0
               Last_SQL_Error: 
  Replicate_Ignore_Server_Ids: 
             Master_Server_Id: 3
               Master_SSL_Crl: 
           Master_SSL_Crlpath: 
                   Using_Gtid: No
                  Gtid_IO_Pos: 
1 row in set (0.00 sec)
```

**Review by LaCap — APPROVED** [review]

**Comment by jespada** [comment]
> show slave status\G

@LaCap  this is for future reference, you can clean up all that config with `stop slave` and `reset slave all`, as can be seen on this gist:
https://gist.github.com/jespada/70b076058f7e77c8964a424ce1eef7bf

**Pull request merged**